From e9bb1036f8efcf6bacb33978c23e8aac4b2951da Mon Sep 17 00:00:00 2001
From: Liu Ying <victor.liu@nxp.com>
Date: Thu, 20 Apr 2017 15:27:57 +0800
Subject: [PATCH 1713/5242] MLK-14718 video: fbdev: mxc: mipi dsi: Check
 device tree node in probe

commit  35beac2593260da0ae2e9c7ca8317b33acff843d from
https://source.codeaurora.org/external/imx/linux-imx.git

For the safe probing's sake, check valid device tree node in probe function.

Signed-off-by: Liu Ying <victor.liu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/video/fbdev/mxc/mipi_dsi.c |    5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/video/fbdev/mxc/mipi_dsi.c b/drivers/video/fbdev/mxc/mipi_dsi.c
index 4e189cd..d0f064e 100644
--- a/drivers/video/fbdev/mxc/mipi_dsi.c
+++ b/drivers/video/fbdev/mxc/mipi_dsi.c
@@ -839,6 +839,11 @@ static int mipi_dsi_probe(struct platform_device *pdev)
 	int mux;
 	int ret = 0;
 
+	if (!np) {
+		dev_err(&pdev->dev, "failed to find device tree node\n");
+		return -ENODEV;
+	}
+
 	mipi_dsi = devm_kzalloc(&pdev->dev, sizeof(*mipi_dsi), GFP_KERNEL);
 	if (!mipi_dsi)
 		return -ENOMEM;
-- 
1.7.9.5

